### PR TITLE
chore: cleanup CODEOWNERS file

### PR DESCRIPTION
**Pull request by Shabirmean**
#### Description
- This PR is an audit of the CODEOWNERS file
- The audit is done in light of the Q2 samples fixit

#### Changes
- Add comments at the top with instructions for updating the CODEOWNERS file and link to [go/jds-codeowners](http://go/jds-codeowners)
- Set up indentation such that it is easy to read and visualize the teams
- Update entry for `appengine` to be `appengine-*`
- Update `error-reporting` to `errorreporting`
- Remove duplicate entry for `datacatalog`
- Remove following entries given that there are no samples for them in the repo
   - scheduler
   - orgpolicy
   - container
   - game-servers
   - logging
   - datastore
   - firestore
   - notebooks
   - dataplex
- Add Github team @GoogleCloudPlatform/healthcare-life-sciences as codeowner for `healthcare` instead of an IC
- Commit https://github.com/GoogleCloudPlatform/java-docs-samples/pull/7835/commits/547a51bfe4a4d378e76c10be09f1370378c6dadc updates the PR with changes from #7836

### CODE REVIEW

**Review by averikitsch — APPROVED** [review]
Lgtm, but I am wondering if we need to preventively add a prefix `/` to signify root directories https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file

File: .github/CODEOWNERS
Comment (rsamborski) [inline .github/CODEOWNERS:10]: As I am working on a horizontal view for all languages, can I suggest to use go/dee-codeowners (WIP) so we can have a single source of truth for all DEE owned languages?

File: .github/CODEOWNERS
Comment (rsamborski) [inline .github/CODEOWNERS:14]: ditto - can we switch to common go/dee-codeowners?

**Review by Sita04 — APPROVED** [review]

File: .github/CODEOWNERS
Comment (Shabirmean) [reply .github/CODEOWNERS]: Changed this for now. But there are multiple directories that we find in this repo - missing in go/dee-codeowners. I think this will prevent users from easily finding the correct OWNERS.

File: .github/CODEOWNERS
Comment (Shabirmean) [reply .github/CODEOWNERS]: See note above. I'd like to revisit this later. But changing it for the purpose of merging this PR

**Comment by Shabirmean** [comment]
> Lgtm, but I am wondering if we need to preventively add a prefix `/` to signify root directories https://docs.github.com/en/repositories/managing-your-repositorys-settings-and-features/customizing-your-repository/about-code-owners#example-of-a-codeowners-file

Makes sense. I added the forward slashes!

**Review by rsamborski — APPROVED** [review]
@Shabirmean Thanks for updating. As mentioned on chat I am working on updating go/dee-codeowners and it cover all missing folders by end of next week.

**Pull request merged**
